src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ObjectToIdCache.cs

```diff
@@ -175,8 +175,7 @@ private static int GetPrime(int min)
             3, 7, 17, 37, 89, 197, 431, 919, 1931, 4049, 8419, 17519, 36353,
             75431, 156437, 324449, 672827, 1395263, 2893249, 5999471,
             11998949, 23997907, 47995853, 95991737, 191983481, 383966977, 767933981, 1535867969,
-            2146435069, 0X7FEFFFFF
-            // 0X7FEFFFFF is not prime, but it is the largest possible array size. There's nowhere to go from here.
+            2146435069, Array.MaxLength // Array.MaxLength is not prime, but it is the largest possible array size. There's nowhere to go from here.
```

Comment:
Is this going to turn off the efficient pre-ininitialization path for this array?

Reply:
Ah, probably.
```suggestion
            2146435069, 0x7FFFFFC7
            // 0x7FFFFFC7 == Array.MaxLength is not prime, but it is the largest possible array size.
            // There's nowhere to go from here. Using a const rather than the MaxLength property
            // so that the array contains only const values.
```